File: f822 紅林愛下棋.py
from sys import exit
final = [list('R' * 11)] + [list('R' + input() + 'R') for i in range(9)] + [list('R' * 11)]
def count_it(x, y):
	d = {'.':0,'W':0,'B':0,'R':0}
	d[final[x][y - 1]] += 1
	d[final[x][y + 1]] += 1
	d[final[x - 1][y]] += 1
	d[final[x + 1][y]] += 1
	return d['.'], d['W'], d['B']
for i in range(16):
	for x in range(1, 10):
		for y in range(1, 10):
			if final[x][y] == '.':
				dot, W, B = count_it(x, y)
				if W == 0 and B == 0:continue
				elif W != 0 and B != 0:
					print("Wrong")
					exit()
				final[x][y] = 'W' if B == 0 else 'B'

final_str = ''
for x in range(1, 10):
	for y in range(1, 10):
		final_str += final[x][y]
	final_str += '\n'

W = final_str.count('W')
B = 81 - W
print("Black wins!!\n" if B > W else "White wins!!\n")
print(f"{B}:{W}\n")
print(final_str)

File: f822 紅林愛下棋.cpp
#include <bits/stdc++.h>
using namespace std;

void print(int a, int b, int board1[11][11], int BW){
    if(board1[a+1][b]=='.') {board1[a+1][b]=BW; print(a+1,b,board1,BW);}
    if(board1[a][b+1]=='.') {board1[a][b+1]=BW; print(a,b+1,board1,BW);}
    if(board1[a][b-1]=='.') {board1[a][b-1]=BW; print(a,b-1,board1,BW);}
    if(board1[a-1][b]=='.') {board1[a-1][b]=BW; print(a-1,b,board1,BW);}
}

int main()
{
    int board[11][11], board_W[11][11], board_B[11][11] ;
    for(int i=0; i<11; i++)
        for(int j=0; j<11; j++){board[i][j]='S'; board_W[i][j]='S'; board_B[i][j]='S';}
    for(int i=1; i<10; i++){
        for(int j=1; j<10; j++){board[i][j]=getchar(); board_W[i][j]=board[i][j]; board_B[i][j]=board[i][j];}
        getchar() ;
    }
    //for(int i=0; i<11; i++){for(int j=0; j<11; j++)printf("%c ", board[i][j]) ;printf("\n") ;}
    for(int i=1; i<10; i++)
        for(int j=1; j<10; j++){
            if(board[i][j]=='B') print(i,j,board_B,'B') ;
            if(board[i][j]=='W') print(i,j,board_W,'W') ;
        }
    //for(int i=0; i<11; i++){for(int j=0; j<11; j++)printf("%c ", board_W[i][j]) ;printf("\n") ;}
    //for(int i=0; i<11; i++){for(int j=0; j<11; j++)printf("%c ", board_B[i][j]) ;printf("\n") ;}
    int black=0, white=0 ;
    for(int i=1; i<10; i++)
        for(int j=1; j<10; j++){
            if(board_W[i][j]=='W'){white++; board[i][j]='W';}
            if(board_B[i][j]=='B'){black++; board[i][j]='B';}
            if(board_W[i][j]=='W'&&board_B[i][j]=='B'){printf("Wrong"); return 0;}
        }
    printf((black>white)?"Black wins!!\n":"White wins!!\n") ;
    printf("%d:%d\n", black, white) ;
    for(int i=1; i<10; i++){for(int j=1; j<10; j++)printf("%c", board[i][j]) ;printf("\n") ;}
    return 0;
}